tests/unit_tests/test_tethys_cli/test_proxyapps_commands.py

```diff
@@ -0,0 +1,223 @@
+        mock_args = mock.Mock()
+        mock_args.verbose = True
+        list_proxyapps(mock_args)
+        rts_call_args = mock_print.call_args_list
+        # breakpoint()
+
+        expected_output = (
+            f"  {self.app_name}:\n"
+            f"    endpoint: {self.endpoint}\n"
+            f"    description: {self.description}\n"
+            f"    logo_url: {self.logo}\n"
+            f"    tags: {self.tags}\n"
+            f"    enabled: {self.enabled}\n"
+            f"    show_in_apps_library: {self.show_in_apps_library}\n"
+            f"    back_url: {self.back_url}\n"
+            f"    open_in_new_tab: {self.open_in_new_tab}\n"
+            f"    display_external_icon: {self.display_external_icon}\n"
+            f"    order: {self.order}"
+        )
+
+        mock_write_info.assert_called_with("Proxy Apps:")
+        self.assertEqual(rts_call_args[0][0][0], expected_output)
+
+    @mock.patch("tethys_cli.proxyapps_commands.write_error")
+    @mock.patch("tethys_cli.proxyapps_commands.exit", side_effect=SystemExit)
+    def test_update_proxy_apps_no_app(self, mock_exit, mock_write_error):
+        mock_args = mock.Mock()
+        mock_args.name = "non_existing_proxy_app"
+        mock_args.set_kwargs = [["non_existing_key", "https://fake.com"]]
+
+        # mock_args.proxy_app_key = "non_existing_key"
+        # mock_args.proxy_app_key_value = "https://fake.com"
+
+        self.assertRaises(
+            SystemExit,
+            update_proxyapp,
+            mock_args,
+        )
+
+        mock_write_error.assert_called_with(
+            "Proxy app named 'non_existing_proxy_app' does not exist"
+        )
+        mock_exit.assert_called_with(1)
+
+    @mock.patch("tethys_cli.proxyapps_commands.write_success")
+    @mock.patch("tethys_cli.proxyapps_commands.write_warning")
+    @mock.patch("tethys_cli.proxyapps_commands.exit", side_effect=SystemExit)
+    def test_update_proxy_apps_no_correct_key(
+        self, mock_exit, mock_write_warning, mock_write_success
+    ):
+        mock_args = mock.Mock()
+        mock_args.name = self.app_name
+        mock_args.set_kwargs = [["non_existing_key", "https://fake.com"]]
+        mock_args.proxy_app_key = "non_existing_key"
+        mock_args.proxy_app_key_value = "https://fake.com"
+
+        self.assertRaises(
+            SystemExit,
+            update_proxyapp,
+            mock_args,
+        )
+
+        mock_write_warning.assert_called_with(
+            "Attribute non_existing_key does not exist"
+        )
+        mock_write_success.assert_called_with(
+            f"Proxy app '{self.app_name}' was updated successfully"
+        )
+        mock_exit.assert_called_with(0)
+
+    @mock.patch("tethys_cli.proxyapps_commands.write_info")
+    @mock.patch("tethys_cli.proxyapps_commands.write_success")
+    @mock.patch("tethys_cli.proxyapps_commands.exit", side_effect=SystemExit)
+    def test_update_proxy_apps(self, mock_exit, mock_write_success, mock_write_info):
```

Comment:
Since we are using the database in this test, it would be good to run a query and get the db objects after runing the update command to verify that the changes have been saved in the database.